java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/analyser/build/LookupBuildInfoCommand.java

```diff
@@ -131,36 +134,42 @@ public class LookupBuildInfoCommand implements Runnable {
 
     @Override
     public void run() {
+        String infoString;
         try {
             ScmInfo scmInfo = new ScmInfo("git", this.scmUrl);
             Log.infof("LookupBuildInfo resolving %s for version %s ", scmInfo.getUri(), version);
 
-            CacheBuildInfoLocator buildInfoLocator = RestClientBuilder.newBuilder().baseUri(new URI(cacheUrl))
+            if (buildRecipePath != null) {
+                Log.infof("Using build recipe from %s", buildRecipePath);
+                infoString = Files.readString(Path.of(buildRecipePath));
+                Log.infof("Reading %s", infoString);
+            } else {
+                CacheBuildInfoLocator buildInfoLocator = RestClientBuilder.newBuilder().baseUri(new URI(cacheUrl))
                     .build(CacheBuildInfoLocator.class);
-            BuildRecipeInfo buildRecipeInfo = buildInfoLocator.resolveBuildInfo(scmInfo.getUri(), version);
```

Comment:
The infoString should be deserialized into a BuildRecipeInfo, and then used instead of this lookup. I.e. rather than skipping all the logic if it is provided, in should just replace what is looked up from the git repo.